### PR TITLE
LPS-169696 Add feature flag

### PR DESCRIPTION
**Pull request by marcelabc**
Related issue: https://issues.liferay.com/browse/LPS-169696

### CODE REVIEW

**Comment by marcelabc** [comment]
ci:test:sf

**Comment by marcelabc** [comment]
ci:test:sf

**Comment by marcelabc** [comment]
ci:test:relevant

File: modules/apps/notification/notification-rest-api/src/main/java/com/liferay/notification/rest/dto/v1_0/util/NotificationUtil.java [hunk 1]
```diff
@@ -99,6 +100,13 @@ public static NotificationTemplate toNotificationTemplate(
 					0L);
 		}
 
+		if (GetterUtil.getBoolean(PropsUtil.get("feature.flag.LPS-168220")) &&
+			(notificationTemplateId != 0)) {
```
Comment (carolmariaabb) [inline modules/apps/notification/notification-rest-api/src/main/java/com/liferay/notification/rest/dto/v1_0/util/NotificationUtil.java:104]: @marcelabc why do we need this condition `notificationTemplateId != 0`?

File: modules/apps/notification/notification-rest-impl/src/main/java/com/liferay/notification/rest/internal/resource/v1_0/NotificationTemplateResourceImpl.java [hunk 2]
```diff
@@ -141,9 +142,22 @@ public NotificationTemplate postNotificationTemplate(
...
+				notificationTemplate.getExternalReferenceCode())) {
+
+			serviceBuilderNotificationTemplate =
+				_notificationTemplateService.updateExternalReferenceCode(
+					serviceBuilderNotificationTemplate.
+						getNotificationTemplateId(),
+					notificationTemplate.getExternalReferenceCode());
+		}
+
+		return _toNotificationTemplate(serviceBuilderNotificationTemplate);
```
[the remark this answers is not in the document]
Comment (carolmariaabb) [reply modules/apps/notification/notification-rest-impl/src/main/java/com/liferay/notification/rest/internal/resource/v1_0/NotificationTemplateResourceImpl.java:177]: you didn't add there

**Comment by marcelabc** [comment]
ci:test:sf

**Comment by marcelabc** [comment]
ci:test:relevant

File: modules/apps/notification/notification-service/src/main/java/com/liferay/notification/service/impl/NotificationTemplateLocalServiceImpl.java [hunk 3]
```diff
@@ -189,6 +189,20 @@ public NotificationTemplate getNotificationTemplate(
...
+
+		NotificationTemplate notificationTemplate =
+			notificationTemplatePersistence.findByPrimaryKey(
+				notificationTemplateId);
+
+		notificationTemplate.setExternalReferenceCode(externalReferenceCode);
+
+		return notificationTemplatePersistence.update(notificationTemplate);
+	}
+
```
[the remark this answers is not in the document]
Comment (marcelabc) [reply modules/apps/notification/notification-service/src/main/java/com/liferay/notification/service/impl/NotificationTemplateLocalServiceImpl.java]: I was following the Object Definition pattern, but there was no need.

File: modules/apps/notification/notification-rest-impl/src/main/java/com/liferay/notification/rest/internal/resource/v1_0/NotificationTemplateResourceImpl.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (marcelabc) [reply modules/apps/notification/notification-rest-impl/src/main/java/com/liferay/notification/rest/internal/resource/v1_0/NotificationTemplateResourceImpl.java]: I was following the Object Definition pattern, but there was no need.

File: modules/apps/notification/notification-rest-api/src/main/java/com/liferay/notification/rest/dto/v1_0/util/NotificationUtil.java (hunk 1 above)
Comment (marcelabc) [reply modules/apps/notification/notification-rest-api/src/main/java/com/liferay/notification/rest/dto/v1_0/util/NotificationUtil.java]: I don't know, but I removed this part. Thanks!

**Review by carolmariaabb — APPROVED** [review]

**Comment by carolmariaabb** [comment]
ci:test:relevant

**Comment by marcelabc** [comment]
resend: https://github.com/liferay-objects/liferay-portal/pull/1438

**Pull request closed without merging**